release-process.md

```diff
@@ -39,8 +39,18 @@ If you are a new Release Manager, you can read up on the process from the follow
 
 The main step towards preparing a release is to create a release branch. This is done via
 standard Git branching mechanism and should be announced to the community once the branch is
-created. It is also good to set up Jenkins jobs for the release branch once it is cut to
-ensure tests are passing (consult Josh Rosen and Shane Knapp for help with this).
+created.
+
+It is also good to set up Jenkins jobs for the release branch once it is cut to
+ensure tests are passing. These are jobs like 
+https://amplab.cs.berkeley.edu/jenkins/view/Spark%20QA%20Test/job/spark-branch-2.3-test-maven-hadoop-2.7/ .
+Consult Josh Rosen and Shane Knapp for help with this. Also remember to add the newly-added jobs
+to the test dashboard at https://amplab.cs.berkeley.edu/jenkins/view/Spark%20QA%20Test%20(Dashboard)/ .
+
+Finally, after a new branch is cut, update SNAPSHOT versions found in the `master` branch to the
+next minor version. For example when cutting `branch-2.3`, update `master` branch versions from
+`2.3.0-SNAPSHOT` to `2.4.0-SNAPSHOT` in `pom.xml` files and update files like `R/pkg/DESCRIPTION`
+to refer to `2.4.0` rather than `2.3.0`.
```

Comment:
@srowen, I think we should change `__version__` in `python/pyspark/version.py`, say, `2.3.0.dev0` -> `2.4.0.dev0` (reminder - https://github.com/apache/spark/commit/9a145fd796145d1386fd75c01e4103deadb97ac9)

Reply:
as note in the other PR, version changes are done automatically by `dev/create-release/release-tag.sh`? (which is in the `Cutting a Release Candidate` below)

Reply:
Sorry for the late reply. OK @felixcheung you're saying this is (or should be) done by the script? I'll just revert this part.